src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -784,27 +788,67 @@ int fx_muxer_t::execute(const int argc, const pal::char_t* argv[])
         trace::error(_X("Failed to resolve full path of the current executable [%s]"), own_path.c_str());
         return StatusCode::LibHostCurExeFindFailure;
     }
+
     pal::string_t own_name = get_filename(own_path);
     pal::string_t own_dir = get_directory(own_path);
-
     pal::string_t own_dll_filename = get_executable(own_name) + _X(".dll");
     pal::string_t own_dll = own_dir;
     append_path(&own_dll, own_dll_filename.c_str());
 
-    trace::info(_X("Own DLL path=[%s]"), own_dll.c_str());
-    auto mode = detect_operating_mode(own_dir, own_dll, own_name);
+    trace::info(_X("Own dll path '%s'"), own_dll.c_str());
+
+    // Flag to indicate activation of an app instead of an invocation of the SDK.
     bool is_an_app = true;
+
+    // Detect invocation mode
+    auto mode = detect_operating_mode(own_dir, own_dll, own_name);
+
+    //
+    // Invoked as corehost
+    //
+
     if (mode == host_mode_t::split_fx)
     {
         trace::verbose(_X("--- Executing in split/FX mode..."));
         return parse_args_and_execute(own_dir, own_dll, 1, argc, argv, false, host_mode_t::split_fx, &is_an_app);
     }
+
+    //
+    // Invoked from the application base.
+    //
+
     if (mode == host_mode_t::standalone)
     {
         trace::verbose(_X("--- Executing in standalone mode..."));
-        return parse_args_and_execute(own_dir, own_dll, 1, argc, argv, false, host_mode_t::standalone, &is_an_app);
+
+        bool is_unsigned_host = exe_type.empty();
+        bool is_app_host = (exe_type == _X("apphost"));
+        bool is_dotnet_host = (exe_type == _X("dotnet"));
+        bool can_load_own_dll = is_unsigned_host || (is_app_host && pal::validate_binding(own_dll));
+
+        trace::info(_X("Activation parameters, unsigned host: '%d', app host: '%d', loadable: '%d'"), is_unsigned_host, is_app_host, can_load_own_dll);
+
+        // Temporarily allow "dotnet.exe" host to load by own name.
```

Comment:
Is this being done for transition to this change by other components? If so, I would suggest enabling it behind an env var so that it cannot be accidentally used.